ddtrace/span.py

```diff
@@ -372,6 +382,9 @@ def set_exc_info(self, exc_type, exc_val, exc_tb):
         if not (exc_type and exc_val and exc_tb):
             return  # nothing to do
 
+        if self._ignored_exceptions and exc_type in self._ignored_exceptions:
```

Comment:
I'm on the fence about whether we want to do a direct comparison (`exc_type in self._ignored_exceptions`) or do an `isinstance` check on each ignored exception.

Reply:
Decided with @majorgreys to go ahead with the isinstance check to handle cases when users might subclass the exceptions that we ignore